### PR TITLE
feature(yzj): add scatter connection model

### PR DESCRIPTION
**Pull request by jayyoung0802**
## Description
add scatter connection model (Scatter feature to its corresponding location)

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/torch_utils/network/scatter_connection.py [hunk 1]
```diff
@@ -91,3 +91,51 @@ def forward(self, x: torch.Tensor, spatial_size: Tuple[int, int], location: torc
         output = output.reshape(N, B, H, W)
         output = output.permute(1, 0, 2, 3).contiguous()
         return output
+    
+    @hpc_wrapper(
```
Comment (PaParaZz1) [inline ding/torch_utils/network/scatter_connection.py:95]: you don't need to add hpc wrapper here

File: ding/torch_utils/network/tests/test_scatter.py [hunk 2]
```diff
@@ -29,3 +29,18 @@ def test_naive(self):
...
+            SpatialSize = (13, 17)
+            input = torch.randn(size=(BatchSize, Num, EmbeddingSize)).requires_grad_(True)
+            coord_x = torch.randint(low=0, high=13, size=(BatchSize, Num))
+            coord_y = torch.randint(low=0, high=17, size=(BatchSize, Num))
+            output = model.xy_forward(input, SpatialSize, coord_x, coord_y)
+            loss = output.mean()
+            loss.backward()
+            assert isinstance(input.grad, torch.Tensor)
+            assert output.shape == (BatchSize, EmbeddingSize, *SpatialSize)
+
```
Comment (PaParaZz1) [inline ding/torch_utils/network/tests/test_scatter.py:46]: remove empty line at the end

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/torch_utils/network/tests/test_scatter.py (hunk 2 above)
Comment (jayyoung0802) [reply ding/torch_utils/network/tests/test_scatter.py]: done

File: ding/torch_utils/network/scatter_connection.py (hunk 1 above)
Comment (jayyoung0802) [reply ding/torch_utils/network/scatter_connection.py]: done

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**